src/Magento/FunctionalTestingFramework/ObjectManager/Config/Config.php

```diff
@@ -207,32 +207,44 @@ protected function mergeConfiguration(array $configuration)
                     break;
 
                 default:
-                    $key = ltrim($key, '\\');
-                    if (isset($curConfig['type'])) {
-                        $this->virtualTypes[$key] = ltrim($curConfig['type'], '\\');
-                    }
-                    if (isset($curConfig['arguments'])) {
-                        if (!empty($this->mergedArguments)) {
-                            $this->mergedArguments = [];
-                        }
-                        if (isset($this->arguments[$key])) {
-                            $this->arguments[$key] = array_replace($this->arguments[$key], $curConfig['arguments']);
-                        } else {
-                            $this->arguments[$key] = $curConfig['arguments'];
-                        }
-                    }
-                    if (isset($curConfig['shared'])) {
-                        if (!$curConfig['shared']) {
-                            $this->nonShared[$key] = 1;
-                        } else {
-                            unset($this->nonShared[$key]);
-                        }
-                    }
+                    $this->setArguments($key, $curConfig);
```

Comment:
Possible a more accurate method name, e.g. setConfiguration() or setNonPreferenceConfiguration()

Reply:
Not in your change, but i think the "break" next line in 211 is not necessary and should be removed.

Reply:
Fixed.